server/controllers/article.js

```diff
@@ -240,37 +240,55 @@ class ArticleController {
   async getSingleArticle(req, res) {
     const { slug } = req.params;
     try {
-      const article = await Article.findOne({
-        where: {
-          slug
-        },
-        attributes: {
-          exclude: [
-            'id',
-            'userId'
-          ]
-        },
-        include: [
-          {
-            model: User,
-            attributes: ['firstname', 'lastname', 'username', 'email', 'image']
-          },
-          {
-            model: Tag,
-            attributes: ['name']
-          }
-        ]
-      });
+      const article = await this.getArticle(slug);
       const readTime = await HelperUtils.estimateReadingTime(article.body);
       const singleArticle = [article].map((data) => {
         data.dataValues.readTime = readTime;
         return data;
       });
-      response(res).success({ message: singleArticle[0] });
+      if (req.user) {
+        if (req.user.id !== req.article.userId) {
```

Comment:
Why do you have multiple if statement, here

Reply:
Because req.user will be undefined if the user isn't logged in and I can't get the id property of undefined so req.user.id will return an error.

Nevertheless, I will try and find a better way to handle it